src/editor/view/file/git.rs

```diff
@@ -106,93 +108,83 @@ fn get_diff_result(
     content: &str,
     file_path: &str,
     file_name: &str,
-) -> Result<String, Box<dyn std::error::Error>> {
-    // Get the file_name relative to the file_path git repository
-    let file_name = Command::new("git")
-        .current_dir(file_path)
-        .args(["ls-files", "--full-name", file_name])
-        .output()?
-        .stdout;
-    let file_name = String::from_utf8_lossy(&file_name).trim().to_string();
-
-    if file_name.is_empty() {
-        // It is a new file in the git repository so we return the number of lines
-        // as the number of lines added at the beginning of the file
-        return Ok(format!("0a1,{}", content.lines().count().max(1)));
-    }
-
-    // Execute the shell command
-    // It would be better to not rely on bash but I don't know how to emulate the process substitution
-    // with Rust.
-    let mut diff = Command::new("bash")
-        .current_dir(file_path)
-        .stdin(Stdio::piped())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .arg("-c")
-        .arg(format!("diff <(git show HEAD:{}) -", file_name))
-        .spawn()?;
-
-    // Write the content to diff on the stdin of the process
-    let diff_input = diff.stdin.as_mut().unwrap();
-    diff_input.write_all(content.as_bytes())?;
-
-    // Wait for the process to finish
-    let diff_output = diff.wait_with_output()?;
-
-    // Check the exit code and parse stdout/stderr accordingly
-    let status_code = diff_output.status.code();
-    if matches!(status_code, Some(0 | 1)) {
-        Ok(String::from_utf8(diff_output.stdout)?)
-    } else {
-        Err(String::from_utf8(diff_output.stderr)?.into())
+) -> Result<Vec<Patch>, Box<dyn std::error::Error>> {
+    let old_content = std::fs::read_to_string(file_path.to_owned() + file_name);
```

Comment:
You are taking the content of the file on disk as reference instead of its value at HEAD. Saving the file will therefore reset the diff. You should use git2-rs to get the content of the file in the previous version.